daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -384,6 +384,24 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
       case (acc, _) => acc
     } -- localContracts.keySet
 
+  /** Return all the contract keys created, exercised, fetched or lookup by this transation.
+    */
+  final def contractKeys[Cid2 >: Cid]: Set[(Ref.Identifier, Value[Cid2])] = {
```

Comment:
Should we have a test for this somewhere?

Reply:
fine with merging without tests given that there don’t really seem to be tests for the inputs and effects logic either but we should fix this so maybe add a todo.

Reply:
done